core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -226,6 +240,29 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
       fetcherThreadMap.clear()
     }
   }
+
+  def shutdown(): Unit = {
+    info("shutting down")
+    try {
+      closeAllFetchers()
+    } finally {
+      removeMetrics()
+    }
+    info("shutdown completed")
+  }
+
+  private[server] def removeMetrics(): Unit = {
+    metricNamesToTags.foreach(metricTags => {
```

Comment:
perhaps we also want to remove the metric from metricNamesToTags itself, other that map can be an always increasing map. Same comment for other PRs.

Reply:
updated.